homeassistant/components/modbus/__init__.py

```diff
@@ -69,11 +93,40 @@
 
 BASE_SCHEMA = vol.Schema({vol.Optional(CONF_NAME, default=DEFAULT_HUB): cv.string})
 
-CLIMATE_SCHEMA = vol.Schema(
+
+def number(value: Any) -> Union[int, float]:
+    """Coerce a value to number without losing precision."""
+    if isinstance(value, int):
+        return value
+    if isinstance(value, float):
+        return value
+
+    try:
+        value = int(value)
+        return value
+    except (TypeError, ValueError):
+        pass
+    try:
+        value = float(value)
+        return value
+    except (TypeError, ValueError) as err:
+        raise vol.Invalid(f"invalid number {value}") from err
+
+
+BASE_COMPONENT_SCHEMA = vol.Schema(
     {
-        vol.Required(CONF_CURRENT_TEMP): cv.positive_int,
         vol.Required(CONF_NAME): cv.string,
-        vol.Required(CONF_SLAVE): cv.positive_int,
+        vol.Optional(CONF_SLAVE): cv.positive_int,
```

Comment:
In the climate schema, slave was required, but in the base schema slave is optional. Does that change matter?

Reply:
First of all, thanks for a long review, as usual most of your requests makes sense (there are one, where I need to think about how to do it).

Slave should only be used when we connect to a modbus device, that have modbus devices "below" it, so it was an error to have it as required.